cubedash/_stac.py

```diff
@@ -0,0 +1,235 @@
+    [minimum longitude, minimum latitude, maximum longitude, maximum latitude]
+    """
+
+    query = dict()
+    if product:
+        query['product'] = product
+    time_period = time.split('/')
+    if len(time_period) == 2:
+        query['time'] = Range(Timestamp(time_period[0]), Timestamp(time_period[1]))
+    elif len(time_period) == 1:
+        query['time'] = Range(Timestamp(time_period[0]).floor('1D'),
+                              (Timestamp(time_period[0]) + DateOffset(1)).floor('1D'))
+    else:
+        return []
+
+    # bbox is in GeoJSON CRS (WGS84)
+    query['lon'] = Range(bbox[0], bbox[2])
+    query['lat'] = Range(bbox[1], bbox[3])
+
+    return _model.STORE.index.datasets.search(limit=limit, **query)
+
+
+def stac_dataset(dataset):
+    """
+    Returns a dict corresponding to a stac item catalog
+    """
+    metadata_doc = dataset.metadata_doc_without_lineage()
+    uri = dataset.local_uri
+
+    if metadata_doc['grid_spatial']['projection'].get('valid_data', None):
+        geodata = valid_coord_to_geojson(metadata_doc['grid_spatial']['projection']['valid_data'],
+                                         dataset.crs)
+    else:
+        # Compute geometry from geo_ref_points
+        points = [[list(point.values()) for point in
+                   metadata_doc['grid_spatial']['projection']['geo_ref_points'].values()]]
+
+        # last point and first point should be same
+        points[0].append(points[0][0])
+
+        geodata = valid_coord_to_geojson({'type': 'Polygon', 'coordinates': points}, dataset.crs)
+
+    # Convert the date to add time zone.
+    center_dt = parse(metadata_doc['extent']['center_dt'])
+    center_dt = center_dt.replace(microsecond=0)
+    time_zone = center_dt.tzinfo
+    if not time_zone:
+        center_dt = center_dt.replace(tzinfo=datetime.timezone.utc).isoformat()
+    else:
+        center_dt = center_dt.isoformat()
+
+    # parent? We will have an empty parent for now
+    stac_item = OrderedDict([
+        ('id', metadata_doc['id']),
+        ('type', 'Feature'),
+        ('bbox', [metadata_doc['extent']['coord']['ll']['lon'],
+                  metadata_doc['extent']['coord']['ll']['lat'],
+                  metadata_doc['extent']['coord']['ur']['lon'],
+                  metadata_doc['extent']['coord']['ur']['lat']]),
+        ('geometry', geodata),
+        ('properties', {
+            'datetime': center_dt,
+            'provider': CFG['contact']['name'],
+            'license': CFG['license']['name'],
+            'copyright': CFG['license']['copyright'],
+            'product_type': metadata_doc['product_type'],
+            'homepage': CFG['homepage']
+        }),
+        ('links', [
+            {'href': uri, 'rel': 'self'},
+        ]),
+        ('assets', {
+            band_name: {
+                # "type"? "GeoTIFF" or image/vnd.stac.geotiff; cloud-optimized=true
+                'href': band_data['path'],
+                "required": 'true',
```

Comment:
I have carried this thing over from earlier work. Either it was present in earlier STAC versions or Siva may have added it for some other reason. I have lost track of this may be I will remove it.